packages/agent/src/errors.ts

```diff
@@ -215,19 +215,23 @@ export class CertificateVerificationErrorCode extends ErrorCode {
 export class CertificateTimeErrorCode extends ErrorCode {
   public name = 'CertificateTimeErrorCode';
 
+  public readonly systemTime: Date;
+  public readonly timeDiffMsecs: number;
+
   constructor(
     public readonly maxAgeInMinutes: number,
     public readonly certificateTime: Date,
     public readonly currentTime: Date,
-    public readonly timeDiffMsecs: number,
     public readonly ageType: 'past' | 'future',
   ) {
     super();
     Object.setPrototypeOf(this, CertificateTimeErrorCode.prototype);
+    this.systemTime = new Date();
```

Comment:
Why don't we take the system time from which the "adjusted" current time was derived?

Reply:
> Maybe I should just revert https://github.com/dfinity/agent-js/commit/9f8ed1f71e69eaffcce225f7f7a610388c62dc73?

Reverting would mean we don't touch this piece of code (so it does not attract attention on this PR), but the code already exhibited the exact same issue before this PR. So if there's no clear way forward at this point, let's just create a ticket about this issue and resolve this conversation.

Reply:
Or would it be possible to clean up this piece of code after reverting? (Then I'd be up for reverting and _following up_.)